poet/poet.py

```diff
@@ -76,10 +85,106 @@ def walk(package):
     return sorted(discovered)
 
 
-def research_package(name, version=None):
+class PipSourceMetadataException(Exception):
+    pass
+
+@dataclass
+class PackageMetadata:
+    name: str
+    homepage: str
+    url: str
+    checksum: str
+    checksum_type: str
```

Comment:
```suggestion
    checksum_type: str = "sha256"
```
this is one where i think we can set a sensible default?